### DISCUSSION TITLE
Unimgr OVS Driver Support

### DISCUSSION THREAD

**Discussion by jahveen** [MEF-GIT/MEF-LSO-Presto-SDK#21] [Team Posts]
Hi,

I have recently started using the MEF LSO Presto SDK and have found the SDK to be useful in enabling me to understand the concept of the API and OpenDaylight.

I am attempting to create connectivity services using the SDK (R3), the github fork of OpenDaylight with Unimgr (https://github.com/donaldh/unimgr/tree/presto-20180307), and the OVS Driver, however I find that the OVS Driver is limited. When I specify a CE-VLAN-ID (of which only one is supported), I initially expected that VLAN ID to be used to identify which EVC the ingress traffic belongs to. However, it seems that the CE-VLAN-ID is used for an internal vlan id used inside the OvS switch network. Most other attributes have no effect when set.

Am I right in saying that Unimgr has the fully-featured YANG Data Models, but that I must extend the OVS Driver (or any other driver) myself for my own use cases?

Thanks.

Jahveen Davis

**Comment by mariusz-amartus** [comment]
@donaldh, @bartoszm, others,
Could you please help @jahveen and answer his questions?

Thanks,
Mariusz
LSO SDK Community Manager

**Comment by bartoszm** [comment]
Hi @jahveen,
Indeed you are right. Our ovs-driver was used for demo purposes and has a number of limitations. Partially due to Opendaylight modules limitation partially because the functionality was not in the core of our interest.

**Comment by jahveen** [comment]
Thanks for the info. Will release 4 of the SDK resolve any of these limitations? My interest is to be able to perform a good internal demonstration to my colleagues, and having a good foundation would reduce the amount of effort/frustration on my part to deliver it.

**Comment by bartoszm** [comment]
Hi @jahveen,
Uni manager is a project managed as a part of Opendaylight project. Unfortunately, I am not being able to actively participate in the project for some time now. Unimanager project team is always looking for contributions thus please consider whether you can join the effort.
Regardless of your decisions, please contact unimgr-dev@lists.opendaylight.org to verify with the team on the current plans.
